### PR TITLE
add security linter bandit to nox

### LINKED ISSUE #74: ✨ Add security linter to the toolbox
# Summary
Add a security scanner and target to the toolbox features.

## Open Questions

* Should this be mandatory, or should a project be able to enable or disable it?
* Are there other alternatives?
* How and if should this be in the CI? (Projects need to be fixed first)
* How to define exceptions for specific code parts (functions, files) in a project?

## References
* [Bandit](https://github.com/PyCQA/bandit)
* #73 

# Task(s)
```[tasklist]
- [ ] Add bandit as security linter
- [ ] Add a target checking for security issues
- [ ] Update/Add documentation
- [ ] Update changelog
```

### PR DESCRIPTION
**Pull request by Jannis-Mittenzwei**
closes #74

### CODE REVIEW

**Review by ckunki — CHANGES_REQUESTED** [review]

File: test/unit/report_test.py [hunk 1]
```diff
@@ -110,3 +115,42 @@ def test_static_code_analysis(
...
+        ([], 6),
+        (["HL"], ["MH"]),
+        (["MH"], ["MM"]),
+        (["MM"], ["ML"]),
+        (["HL"], ["LL"]),
+        (["LL"], []),
+        (["MH", "LL"], ["MH"]),
+    ],
+)
+def test_bandit_order(lower, compare):
```
Comment (ckunki) [inline test/unit/report_test.py:135]: Proposals:
1. Move auxiliary functions `level()` and `ratings()` to top level, i.e. outside test function `test_bandit_order()`
2. split test cases into a) expecting an `int` b) specifying a different input with better rating 
3. rename parameters appropriately, e.g. `input` and `expected` vs. `lower` and `higher`
4. remove case `assert False`

**Review by ckunki — DISMISSED** [review]

File: .security.json
Comment (Nicoretti) [inline .security.json:1]: should this file be checked in? Likely it is a candidate for the `.gitignore`

**Review by Nicoretti — COMMENTED** [review]
Would be nice if we could add some basic information  regarding the security linter to the doc e.g. where to configure the output ...

**Review by Nicoretti — APPROVED** [review]

File: exasol/toolbox/nox/_lint.py [hunk 2]
```diff
@@ -78,7 +78,7 @@ def type_check(session: Session) -> None:
     _type_check(session, py_files)
 
 
-@nox.session(name="security", python=False)
+@nox.session(name="security-check", python=False)
```
Comment (Nicoretti) [inline exasol/toolbox/nox/_lint.py:81]:
```suggestion
@nox.session(name="security", python=False)
```

**Comment by ckunki** [comment]
Today I had the opportunity together with @Nicoretti to take a look at your PR. 
Nico and I are aware of some changes described in issue #181.
And we propose to wait until #181 is implemented before finalizing the current PR.

File: exasol/toolbox/nox/_lint.py [hunk 3]
```diff
@@ -49,3 +76,10 @@ def type_check(session: Session) -> None:
     """Runs the type checker on the project"""
     py_files = [f"{file}" for file in python_files(PROJECT_CONFIG.root)]
     _type_check(session, py_files)
+
+
+@nox.session(name="security", python=False)
+def security_lint(session: Session) -> None:
+    """runs the security linter bandit on the project without the test files"""
```
Comment (Nicoretti) [inline exasol/toolbox/nox/_lint.py:83]:
```suggestion
    """Runs the security linter on the project"""
```

**Review by Nicoretti — APPROVED** [review]
Looks good!

**Review by ckunki — APPROVED** [review]

**Pull request merged**